### PR TITLE
[WIP]Add vpc documentation

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**
Adds documentation for vpc with examples.

### CODE REVIEW

File: docs/src/analysis/vpc.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+`vpc` computes the quantiles for VPC for a `FittedPumasModel` with simulated prediction intervals 
+around the empirical quantiles based on `reps` simulated populations.
+
+The number of repetitions of simulation are passed in with the `reps` positional argument. 
+
+The following keyword arguments are supported:
+
+    - `quantiles::NTuple{3,Float64}`: A three-tuple of the quantiles for which the quantiles will be computed. The default is `(0.1, 0.5, 0.9)` which computes the 10th, 50th and 90th percentile.
+    - `level::Real`: Probability level to use for the simulated prediction intervals. The default is `0.95`.
+    - `dv::Symbol`: The name of the dependent variable to use for the VPCs. The default is the first dependent variable in the dataset.
```
Comment (vjd) [inline docs/src/analysis/vpc.md:29]:
```suggestion
    - `dv::Symbol`: The name of the dependent variable to use for the VPCs, passed in as a `Symbol`. The default is the first dependent variable in the dataset.
```

File: docs/src/analysis/vpc.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
+# Visual Predictive Check (VPC)
+
+Pumas allows you to generate VPC quantiles for your data and model simulations
+and utilize the Julia plotting capabilities to generate the relevant VPC plots.
+This is allowed with the `vpc` function discuseed below that returns a `VPC` object.
+
```
Comment (vjd) [inline docs/src/analysis/vpc.md:6]: Please mention that we use the quantile regression method as described in the paper and provide a link to that reference

Comment (vjd) [reply docs/src/analysis/vpc.md:6]: and that this is the only way we are doing this

File: docs/src/analysis/vpc.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+```julia
+ vpc_fpm_stratwt = vpc(res, 100, stratify_by=[:wt], bandwidth = 5)
+ plot(vpc_fpm_stratwt)
+```
+![WT VPC](../assets/vpc/vpcstratwt.png)
+
+!!! note
+
+    For most users the method used in quantile regression is not going to be of concern, but if you see large run times switching `qreg_method` to `IP(true)` after loading QuantileRegressions.jl should help in improving the performance with a tradeoff in the accuracy of the regression fitting.
```
Comment (vjd) [inline docs/src/analysis/vpc.md:182]: Finish this section by saying that VPC's for categorical data are WIP and will be made available in the next release.

File: docs/src/analysis/vpc.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+![Bandwidth 2](../assets/vpc/bandwidth2.png)
+
+```julia
+ vpc_data_stratwt = vpc(data, stratify_by=[:wt], bandwidth = 5)
+ plot(vpc_data_stratwt)
+```
+![Bandwidth 5](../assets/vpc/bandwidth5.png)
+
+We see the plot with higher `bandwidth` better captures the data, so we would use the same `bandwidth` value
+for VPC stratified on `wt`.
```
Comment (vjd) [inline docs/src/analysis/vpc.md:163]:
```suggestion
for VPC of the fitted model, stratified on `wt`.  
```

**Comment by Vaibhavdixit02** [comment]
>It would be nice if you can show how to pass in custom obstimes that is different than the observed data? Also, why is that not in the signature?

I assumed based on some past discussion that it wasn't to be documented for the users. I'll add it.

>Why are you not showing other methods, such as doing a vpc based on the model instead of fpm

This is following the convention similar to other diagnostics functions that we don't expose the `PumasModel` dispatches. Since Joga also mentioned vpc not making sense out of context of a fitted model I decided to go forward with only this signature in the docs, we can add if you feel it's a good idea?

**Comment by andreasnoack** [comment]
>>    Why are you not showing other methods, such as doing a vpc based on the model instead of fpm

> This is following the convention similar to other diagnostics functions that we don't expose the PumasModel dispatches. Since Joga also mentioned vpc not making sense out of context of a fitted model I decided to go forward with only this signature in the docs, we can add if you feel it's a good idea?

I agree that it might be better to only document the method for fitted models. At least until we have discussed this in more detail.

File: docs/src/analysis/vpc.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+ vpc_fpm = vpc(res, 100)
+ plot(vpc_fpm)
+```
+![FPM VPC](../assets/vpc/vpcfpm.png)
+
+```julia
+ vpc_fpm_stratwt = vpc(res, 100, stratify_by=[:wt], bandwidth = 5)
+ plot(vpc_fpm_stratwt)
+```
+![WT VPC](../assets/vpc/vpcstratwt.png)
```
Comment (vjd) [inline docs/src/analysis/vpc.md:187]: For this plot, I would suggest using a version with the `legend = false`

File: docs/src/analysis/vpc.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+ write("sim_quantiles.csv", vpc_fpm.simulated_quantiles)
+```
+
+While plotting the obtained `VPC` object with `plot` the following keyword arguments allow the option 
+to include or exclude various components with `true` or `false` respectively:
+
+    - `observations`: Scatter plot of the true observations.
+    - `simquantile_medians`: The median quantile regression of each quantile from the simulations.
+    - `observed_quantiles`: The quantile regressions for the true observations.
+    - `ci_bands`: Shaded region between the upper and lower confidence levels of each quantile from the simulations.
```
Comment (andreasnoack) [inline docs/src/analysis/vpc.md:77]:
```suggestion
    - `ci_bands`: Shaded region between the upper and lower confidence bounds of each quantile from the simulations.
```

File: docs/src/analysis/vpc.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+ using CSV
+ write("sim_quantiles.csv", vpc_fpm.simulated_quantiles)
+```
+
+While plotting the obtained `VPC` object with `plot` the following keyword arguments allow the option 
+to include or exclude various components with `true` or `false` respectively:
+
+    - `observations`: Scatter plot of the true observations.
+    - `simquantile_medians`: The median quantile regression of each quantile from the simulations.
+    - `observed_quantiles`: The quantile regressions for the true observations.
```
Comment (andreasnoack) [inline docs/src/analysis/vpc.md:76]:
```suggestion
    - `observed_quantiles`: The quantile regressions for the real observations.
```

File: docs/src/analysis/vpc.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+ vpc_fpm.popvpc.data_quantiles
+
+ using CSV
+ write("sim_quantiles.csv", vpc_fpm.simulated_quantiles)
+```
+
+While plotting the obtained `VPC` object with `plot` the following keyword arguments allow the option 
+to include or exclude various components with `true` or `false` respectively:
+
+    - `observations`: Scatter plot of the true observations.
```
Comment (andreasnoack) [inline docs/src/analysis/vpc.md:74]:
```suggestion
    - `observations`: Scatter plot of the real observations.
```

File: docs/src/analysis/vpc.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+ plot(vpc_fpm, observations = true)
+ plot(vpc_fpmstratwt, observations = true, ci_bands = false)
+```
+
+### Full examples: 
+
+Let's take a look at a complete example, first we define the model and generate round-trip data.
+
+```julia
+ model = @model begin
```
Comment (andreasnoack) [inline docs/src/analysis/vpc.md:97]: Please fix the indentation of the core here

**Comment by Vaibhavdixit02** [comment]
Should be good to go in now

File: docs/src/analysis/vpc.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
+# Visual Predictive Check (VPC)
+
+Pumas allows you to generate VPC quantiles for your data and model simulations
+and utilize the Julia plotting capabilities to generate the relevant VPC plots.
+This is allowed with the `vpc` function discussed below that returns a `VPC` object.
+We generate the VPCs with the non-parameteric quantile regression approach discussed
```
Comment (pkofod) [inline docs/src/analysis/vpc.md:6]:
```suggestion
We generate the VPCs with the non-parametric quantile regression approach discussed
```

File: docs/src/analysis/vpc.md [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+                quantiles::NTuple{3,Float64}=(0.1, 0.5, 0.9),
+                level::Real=0.95,
+                ensemblealg=EnsembleSerial(),
+                bandwidth=2,
+                numstrats=stratify_by === nothing ? nothing : [4 for i in 1:length(stratify_by)])
+```
+`vpc` computes the quantiles for VPC for a `FittedPumasModel` with simulated confidence intervals 
+around the empirical quantiles based on `reps` simulated populations.
+
+The number of repetitions of simulation are passed in with the `reps` positional argument.
```
Comment (pkofod) [inline docs/src/analysis/vpc.md:25]:
```suggestion
The number of repetitions of the simulations are passed in with the `reps` positional argument. 
```

File: docs/src/analysis/vpc.md [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+    - `dv::Symbol`: The name of the dependent variable to use for the VPCs, passed in as a `Symbol`. The default is the first dependent variable in the dataset.
+    - `stratify_by`: The covariates to be used for stratification. Takes an array of the `Symbol`s of the stratification covariates.
+    - `ensemblealg`: This is passed to the `simobs` call while the `reps` simulations. For more description check the docs for `simobs`.
+    - `bandwidth`: The kernel bandwidth in the quantile regression. If you are seeing `NaN`s or an error, increasing the bandwidth should help in most cases. With higher values of the `bandwidth` you will get more smoothened plots of the quantiles so it's a good idea to check with your data the right `bandwidth`.
+    - `numstrats`: The number of strata to divide into based on the unique values of the covariate, takes an array with the number of strata for the corresponding covariate passed in `stratify_by`. It takes a default of `4` for each of the covariates.
+    - `idv`: The independent variable used for the VPC, deffaults to `:time`.
```
Comment (pkofod) [inline docs/src/analysis/vpc.md:36]:
```suggestion
    - `idv`: The independent variable used for the VPC, defaults to `:time`. 
```

File: docs/src/analysis/vpc.md [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+
+By specifying the plot and the path the VPC plot can be saved to the disk as below:
+```julia
+p = plot(vpc_fpm)
+savefig(p, "simidvsvpc.png")
+```
+
+!!! note
+
+    For most users the method used in quantile regression is not going to be of concern, but if you see large run times switching `qreg_method` to `IP(true)` after loading QuantileRegressions.jl should help in improving the performance with a tradeoff in the accuracy of the regression fitting.
```
Comment (pkofod) [inline docs/src/analysis/vpc.md:202]:
```suggestion
    For most users, the method used in quantile regression is not going to be of concern, but if you see large run times switching `qreg_method` to `IP(true)` after loading QuantileRegressions.jl should help in improving the performance with a tradeoff in the accuracy of the regression fitting.
```

File: docs/src/analysis/vpc.md [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+
+VPCs for discrete or time to event data is already implemented and undergoing internal testing, these should become available 
+in upcoming releases. Additionally prediction corrected VPCs (pcVPCs) are work in progress and will become available soon.
```
Comment (pkofod) [inline docs/src/analysis/vpc.md:205]:
```suggestion
in upcoming releases. Additionally, prediction corrected VPCs (pcVPCs) are work in progress and will become available soon.
```

**Comment by Vaibhavdixit02** [comment]
Thanks @pkofod

**Pull request merged**